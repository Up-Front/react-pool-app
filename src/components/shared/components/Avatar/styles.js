import styled from 'styled-components';

export const AvatarImage = styled.img`
  margin: 5px;
  width: 40px;
  height: 40px;
  border-radius: 50%;

  border-width: 2px;
  border-style: solid;
  border-color: ${props =>
    props.online
      ? props.theme.statusOnlineColor
      : props.theme.statusOfflineColor};
`;
