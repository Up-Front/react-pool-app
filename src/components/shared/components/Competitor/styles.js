import styled from 'styled-components';
import constants from './../../../shared/constants';

export const CompetitorWrapper = styled.div`
  flex: 1;
`;

export const CompetitorLine = styled.div`
  display: flex;
  flex-direction: ${props =>
    props.align === constants.ALIGNRIGHT ? 'row-reverse' : 'row'};
  align-items: center;
  width: 100%;
`;

export const CompetitorName = styled.div`
  flex: 1;
  color: ${props => (props.hasVote ? 'green' : '')};
  text-align: ${props =>
    props.align === constants.ALIGNRIGHT ? 'right' : 'left'};
`;
