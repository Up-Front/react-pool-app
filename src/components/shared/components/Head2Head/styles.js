import styled from 'styled-components';

export const Head2HeadWrapper = styled.div`
  font-weight: normal;
  font-size: 16px;
`;