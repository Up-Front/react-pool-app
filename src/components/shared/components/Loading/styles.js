import styled, { keyframes } from 'styled-components';

const rotate360 = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

export const Rotate = styled.div`
  display: inline-block;
  animation: ${rotate360} 2s linear infinite;
  padding: 2rem 1rem;
  font-size: 1.2rem;
`;
