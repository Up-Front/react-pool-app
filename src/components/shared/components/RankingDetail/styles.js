import styled from 'styled-components';

export const RankingWrapper = styled.div`
  font-size: 24px;
  font-weight: bold;
`;

export const RankMovement = styled.span`
  position: relative;
  top: -3px;
  padding: 0 5px;
  font-size: ${props => props.theme.smallFont};
  font-weight: normal;
  color: rgba(255,255,255,0.5);
`;

export const RankMovementDirection = styled.span`
  position: relative;
  top: 4px;
  margin-right: 2px;
  display: inline-block;
  opacity: ${props =>
    props.movement === 0 || props.movement === undefined ? '0' : '1'};
  color: ${props =>
    props.movement > 0
      ? props.theme.movementUp
      : props.movement < 0 ? props.theme.movementDown : ''};

  transform-origin: center center;
  transform: ${props =>
    props.movement > 0
      ? 'rotate(-90deg)'
      : props.movement < 0 ? 'rotate(90deg)' : ''};

  font-size: 24px;
`;