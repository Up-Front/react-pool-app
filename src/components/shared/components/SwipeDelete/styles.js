import styled from 'styled-components';

export const SwipeWrapper = styled.div`
  position: relative;
  padding: 0;
  overflow: hidden;
`;

export const DeleteLayer = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
`;

export const SwipeContentWrapper = styled.div`
  position: relative;
  z-index: 2;
  user-select: none;
`;

export const SwipeContent = styled.div`
  position: relative;

  transition-property: ${props =>
    props.deleteDirection === 'left' ||
    props.deleteDirection === 'right' ||
    props.deleteCancel
      ? 'left'
      : ''};
  transition-duration: ${props =>
    props.deleteDirection === 'left' ||
    props.deleteDirection === 'right' ||
    props.deleteCancel
      ? '.5s'
      : ''};
  left: ${props =>
    props.deleteDirection === 'left'
      ? '-100%!important'
      : props.deleteDirection === 'right'
        ? '100%!important'
        : props.deleteCancel ? '0!important' : ''};
`;
