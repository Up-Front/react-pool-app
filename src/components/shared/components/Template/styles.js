import styled from 'styled-components';

export const Wrapper = styled.section`
  display: flex;
  align-items: center;
  flex-direction: column;
  height: 100%;
  width: 100%;
  padding: 0;
  background: ${props => props.theme.background};
  background-image: ${props => props.theme.backgroundImage};
  font-size: ${props => props.theme.fontSize};

  & * {
    box-sizing: border-box;
  }
`;

export const Header = styled.header`
  display: flex;
  justify-content: 'space-between';
  ${props => (props.selfStart ? 'align-self: flex-start;' : '')};
  align-items: center;
  flex-direction: 'row';
  padding: 10px;
  width: 100%;
  height: 50px;
  background-color: ${props => props.theme.headerBackgroundColor};

  & a {
    color: ${props => props.theme.headerFontColor};
  }
`;

export const HeaderLogo = styled.h1``;

export const FullScreen = styled.section`
  flex: 1;
  width: 100%;
  margin: 5px 0;
  overflow-y: scroll;
`;
