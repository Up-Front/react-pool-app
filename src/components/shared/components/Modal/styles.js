import styled from 'styled-components';
import { ClearButtonStyle } from './../../styles';
import { MobileWrapper, iconStyle } from './../../../shared/styles';

export const ModalWrapper = styled(MobileWrapper)`
  position: absolute;
  top: 0;
  display: ${props => (props.open ? 'flex' : 'none')};
  flex-direction: column;
  width: 100vw;
  height: 100vh;
  z-index: 9999;

  background: ${props => props.theme.background};
  background-image: ${props => props.theme.backgroundImage};
`;

export const ModalRemoveButton = ClearButtonStyle.extend`
  position: absolute;
  top: 10px;
  right: 15px;
  width: 20px;
  height: 20px;

  color: ${props => props.theme.fontColor};
  font-size: 20px;
  cursor: pointer;

  &:before {
    ${iconStyle('times')}
  }
`;

export const ModalFooter = styled.div`
  padding: 5px;
  height: 50px;
  line-height: 40px;
  border-top: 1px solid ${props => props.theme.borderColor};
  background-color: ${props => props.theme.headerBackgroundColor};
  text-align: right;
`;

export const ModalHeader = styled.div`
  height: 50px;
  border-bottom: 1px solid ${props => props.theme.borderColor};
  background-color: ${props => props.theme.headerBackgroundColor};
`;

export const ModalBody = styled.div`
  flex: 1;
`;
