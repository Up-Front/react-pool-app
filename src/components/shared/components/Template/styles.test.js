import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { theme } from './../../theme';
import { Wrapper, Header, HeaderLogo, FullScreen } from './styles';

describe('Wrapper component', () => {
  test('renders normally', () => {
    const wrapper = shallow(<Wrapper theme={theme} />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Header component', () => {
  test('renders normally', () => {
    const wrapper = shallow(<Header theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('HeaderLogo component', () => {
  xtest('renders normally', () => {
    const wrapper = shallow(<HeaderLogo theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('FullScreen component', () => {
  test('renders normally', () => {
    const wrapper = shallow(<FullScreen theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });
});
