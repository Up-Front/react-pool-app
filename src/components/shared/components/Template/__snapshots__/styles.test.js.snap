// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreen component renders normally 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.section
    theme={
        Object {
            "background": "#243949",
            "backgroundImage": "linear-gradient(to top, #243949 70%, #517fa4 100%)",
            "borderColor": "#FFF",
            "buttonColor": "DarkOrange",
            "buttonFontColor": "#FFF",
            "contestedMatchBackgroundColor": "red",
            "deleteBackgroundColor": "red",
            "fontColor": "#FFF",
            "fontSize": "16px",
            "headerBackgroundColor": "rgba(0, 0, 0, 0.7)",
            "headerFontColor": "#FFF",
            "matchBackgroundColor": "#243949",
            "movementDown": "#da1418",
            "movementUp": "#66b75b",
            "smallFont": "12px",
            "statusOfflineColor": "orange",
            "statusOnlineColor": "green",
            "statusUnknownColor": "lightgrey",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-htoDjs jIzKkg",
    },
    "ref": null,
    "rendered": null,
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-htoDjs jIzKkg",
      },
      "ref": null,
      "rendered": null,
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header component renders normally 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.header
    theme={
        Object {
            "background": "#243949",
            "backgroundImage": "linear-gradient(to top, #243949 70%, #517fa4 100%)",
            "borderColor": "#FFF",
            "buttonColor": "DarkOrange",
            "buttonFontColor": "#FFF",
            "contestedMatchBackgroundColor": "red",
            "deleteBackgroundColor": "red",
            "fontColor": "#FFF",
            "fontSize": "16px",
            "headerBackgroundColor": "rgba(0, 0, 0, 0.7)",
            "headerFontColor": "#FFF",
            "matchBackgroundColor": "#243949",
            "movementDown": "#da1418",
            "movementUp": "#66b75b",
            "smallFont": "12px",
            "statusOfflineColor": "orange",
            "statusOnlineColor": "green",
            "statusUnknownColor": "lightgrey",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-EHOje gOMkQv",
    },
    "ref": null,
    "rendered": null,
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-EHOje gOMkQv",
      },
      "ref": null,
      "rendered": null,
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Wrapper component renders normally 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(styled.div)
    theme={
        Object {
            "background": "#243949",
            "backgroundImage": "linear-gradient(to top, #243949 70%, #517fa4 100%)",
            "borderColor": "#FFF",
            "buttonColor": "DarkOrange",
            "buttonFontColor": "#FFF",
            "contestedMatchBackgroundColor": "red",
            "deleteBackgroundColor": "red",
            "fontColor": "#FFF",
            "fontSize": "16px",
            "headerBackgroundColor": "rgba(0, 0, 0, 0.7)",
            "headerFontColor": "#FFF",
            "matchBackgroundColor": "#243949",
            "movementDown": "#da1418",
            "movementUp": "#66b75b",
            "smallFont": "12px",
            "statusOfflineColor": "orange",
            "statusOnlineColor": "green",
            "statusUnknownColor": "lightgrey",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "sc-ifAKCX ilxlVI",
      "innerRef": undefined,
      "theme": Object {
        "background": "#243949",
        "backgroundImage": "linear-gradient(to top, #243949 70%, #517fa4 100%)",
        "borderColor": "#FFF",
        "buttonColor": "DarkOrange",
        "buttonFontColor": "#FFF",
        "contestedMatchBackgroundColor": "red",
        "deleteBackgroundColor": "red",
        "fontColor": "#FFF",
        "fontSize": "16px",
        "headerBackgroundColor": "rgba(0, 0, 0, 0.7)",
        "headerFontColor": "#FFF",
        "matchBackgroundColor": "#243949",
        "movementDown": "#da1418",
        "movementUp": "#66b75b",
        "smallFont": "12px",
        "statusOfflineColor": "orange",
        "statusOnlineColor": "green",
        "statusUnknownColor": "lightgrey",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "sc-ifAKCX ilxlVI",
        "innerRef": undefined,
        "theme": Object {
          "background": "#243949",
          "backgroundImage": "linear-gradient(to top, #243949 70%, #517fa4 100%)",
          "borderColor": "#FFF",
          "buttonColor": "DarkOrange",
          "buttonFontColor": "#FFF",
          "contestedMatchBackgroundColor": "red",
          "deleteBackgroundColor": "red",
          "fontColor": "#FFF",
          "fontSize": "16px",
          "headerBackgroundColor": "rgba(0, 0, 0, 0.7)",
          "headerFontColor": "#FFF",
          "matchBackgroundColor": "#243949",
          "movementDown": "#da1418",
          "movementUp": "#66b75b",
          "smallFont": "12px",
          "statusOfflineColor": "orange",
          "statusOnlineColor": "green",
          "statusUnknownColor": "lightgrey",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
